packages/errors/src/json-rpc-error.ts

```diff
@@ -124,6 +125,9 @@ export function getSolanaErrorFromJsonRpcError({ code: rawCode, data, message }:
                 // messages in `@solana/errors` to be actual format strings.
                 errorContext = { __serverMessage: message };
                 break;
+            case SOLANA_ERROR__UNRECOGNIZED_JSON_RPC_ERROR:
+                errorContext = { data, message };
+                break;
```

Comment:
This makes sense, but I noticed in the default case we're quite defensive before setting `errorContext = data`: `if (typeof data === 'object' && !Array.isArray(data))`

I guess given you have `data: unknown` we probably don't need that?
Should we maybe just add a string check for message to make sure that type holds? Or we could loosen that type to `unknown` too. Not sure if it's worth worrying about though.

Reply:
Yeah I'm not sure if `message + data` is in the RPC spec but it seems to be the expectation we make on the error response at the moment.

I could just have `message: string` and `errorPayload: unknown` in the error context such that:
- `message` would be equal to `error.message` if it is a there and it's a valid string, otherwise we would fill it with something generic like: `"Unknown RPC error"`.
- `errorPayload` is just the whole `error` object with no assumptions whatsoever on what's inside.